src/MediaWiki/Specials/Admin/DataRepairSection.php

```diff
@@ -0,0 +1,194 @@
+<?php
+
+namespace SMW\MediaWiki\Specials\Admin;
+
+use SMW\ApplicationFactory;
+use SMW\MediaWiki\Renderer\HtmlFormRenderer;
+use Onoi\MessageReporter\MessageReporterFactory;
+use SMW\Message;
+use SMW\MediaWiki\Database;
+use Html;
+use WebRequest;
+use Job;
+use Title;
+
+/**
+ * @license GNU GPL v2+
+ * @since   2.5
+ *
+ * @author mwjames
+ */
+class DataRepairSection {
+
+	/**
+	 * @var Database
+	 */
+	private $connection;
+
+	/**
+	 * @var HtmlFormRenderer
+	 */
+	private $htmlFormRenderer;
+
+	/**
+	 * @var OutputFormatter
+	 */
+	private $outputFormatter;
+
+	/**
+	 * @var boolean
+	 */
+	private $enabledRefreshStore = false;
+
+	/**
+	 * @var null|Job
+	 */
+	private $refreshjob = null;
+
+	/**
+	 * @since 2.5
+	 *
+	 * @param Database $connection
+	 * @param HtmlFormRenderer $htmlFormRenderer
+	 * @param OutputFormatter $outputFormatter
+	 */
+	public function __construct( Database $connection, HtmlFormRenderer $htmlFormRenderer, OutputFormatter $outputFormatter ) {
+		$this->connection = $connection;
+		$this->htmlFormRenderer = $htmlFormRenderer;
+		$this->outputFormatter = $outputFormatter;
+	}
+
+	/**
+	 * @since 2.5
+	 *
+	 * @param boolean $enabledRefreshStore
+	 */
+	public function enabledRefreshStore( $enabledRefreshStore ) {
+		$this->enabledRefreshStore = (bool)$enabledRefreshStore;
+	}
+
+	/**
+	 * @since 2.5
+	 *
+	 * @return string
+	 */
+	public function getForm() {
+
+		$refreshjob = $this->getRefreshJob();
+
+		$this->htmlFormRenderer
+			->setName( 'refreshwiki' )
+			->setMethod( 'post' )
+			->addHiddenField( 'action', 'refreshstore' )
+			->addHeader( 'h2', Message::get( 'smw_smwadmin_datarefresh', Message::TEXT, Message::USER_LANGUAGE ) )
+			->addParagraph( Message::get( 'smw_smwadmin_datarefreshdocu', Message::TEXT, Message::USER_LANGUAGE ) );
+
+		if ( $refreshjob !== null ) {
+			$prog = $refreshjob->getProgress();
+
+			$progressBar = Html::rawElement(
+				'div',
+				array( 'style' => 'float: left; background: #DDDDDD; border: 1px solid grey; width: 300px;' ),
+				Html::rawElement( 'div', array( 'style' => 'background: #AAF; width: ' . round( $prog * 300 ) . 'px; height: 20px; ' ), '' )
+			);
+
+			$this->htmlFormRenderer
+				->addParagraph( Message::get( 'smw_smwadmin_datarefreshprogress' ) )
```

Comment:
>  I get the French version of "smw_smwadmin_datarefreshprogress"

@kghbln Should be `Message::get( 'smw_smwadmin_datarefreshprogress', Message::TEXT, Message::USER_LANGUAGE )`